pkg/ocmagenthandler/ocmagenthandler_service.go

```diff
@@ -17,11 +17,11 @@ import (
 )
 
 func buildOCMAgentService(ocmAgent ocmagentv1alpha1.OcmAgent) corev1.Service {
-	namespacedName := oah.BuildNamespacedName(oah.OCMAgentServiceName)
+	namespacedName := oah.BuildNamespacedName(ocmAgent.Name)
```

Comment:
It is worth noting that this change will cause the OA service/servicemonitor on OSD/ROSA clusters to change from the `ocm-agent` it currently is, to `ocmagent` (which is the name of the current CR).

Therefore I think it could lead to dangling service/servicemonitors lying around on OSD/ROSA clusters when we promote this change, unless we _also_ change the name of the existing `OcmAgent` CR that we push out there. The changing of the name should result in Hive deleting the old `ocmagent` CR from clusters (causing the operator to clean up the service/servicemonitor).

We may want to consider doing that CR name change before merging this (or at least definitely before promoting it)

Reply:
Agree. I think this PR should test along with [OSD-15394](https://issues.redhat.com/browse/OSD-15394) before merging or promoting.